### ISSUE TITLE
getAllPrebidWinningBids() returns something but getAllWinningBids() is empty

### ISSUE THREAD

**Issue by jokatzen** [prebid/Prebid.js#3557]
## Type of issue
Question

## Description
I've been struggling with pbjs and DFP for several days now and my current problem is the one described in the title: when I type `pbjs.getAllPrebidWinningBids()` in the console, something is returned but nothing is displayed on my test page, and when I type `pbjs.getAllWinningBids()`, an empty array is returned and I don't get why.

A few more info :
• This is a test page on our server with no other competition;
• I use custom price buckets;
• In DFP, I have 5 line items from 0.00€ to 2.00€ (so a 0.50€ increment) that matches the custom price buckets in the code;
• The bids are "redirected" in the correct price buckets;
• The code works and an ad is displayed when I set up a self-promotion campaign in DFP with a prebid snippet as a creative, so I suppose that something is wrong with the price buckets.

Do you know what could block the selection of the bid and the rendering of the ad?

## Code
[https://jsfiddle.net/cxen7091/2/](https://jsfiddle.net/cxen7091/2/)

**Comment by jsnellbaker** [comment]
@jokatzen Can you provide a link or a copy of your test page so we can review?

As a brief note to the behavior you're seeing with the two functions:
The `pbjs.getAllPrebidWinningBids()` returns those bids that won the auction, have had their targeting set, but haven't rendered on the page yet.

The `pbjs.getAllWinningBids()` returns those bids that won but also have rendered.

So the behavior you described seems on point; it's just a matter of why the bid isn't rendering on your test page.

**Comment by jokatzen** [comment]
Thanks for your feedback @jsnellbaker, I spent so much time reading the API reference over and over again that I ended up not even getting that!

Here's my code : [](https://jsfiddle.net/cxen7091/2/) (I'll also update my original post).
Also in DFP, I have 5 line items from 0.00€ to 2.00€ (so a 0.50€ increment).

**Comment by jsnellbaker** [comment]
Something to try; the min/max values between the bucket tiers should be same.  Currently you have a dollar gap between the max of one tier and the min of the next tier.  If a bid came back within this gap, it may not be handled properly.

To clarify the functionality of the fields:
- the `max` acts as a hard ceiling; eg a max value of `8` with an increment of `0.05` would mean the highest value accepted into that bucket would be `7.95`.  A bid that was `8.00` would be pushed into the next highest tier (if available).
- the `min` acts as the hard floor; eg a min value of `8` with an increment of `0.05` would mean the lowest value accepted into the bucket would be `8.00`.

The page itself seems to be setup alright otherwise (though you are missing a closing `</body>` tag).  The dummy values for the bidder, DFP setup, etc are making it a little hard to verify.  When I swapped in a test bid from the `appnexus` adapter, the auction went through and the bid was setup to go to DFP.  The DFP call however failed, returning a 400 error.  I'm not sure if that's due to the dummy values or something else.

Please let me know your thoughts on the above and any questions regarding the custom buckets' logic.

**Comment by jokatzen** [comment]
I've modified the values in the bucket tiers (all the bids I receive on this test page are between 0.5€ and 0.9€ so it shouldn't impact much anyway), and it doesn't work either. :(
For the missing `</body>` tag, my bad: it's just a mistake from when I cleaned the code on jsfiddle.

Sorry for the dummy values, it's actually a policy to keep this quite anonymous! However I don't think it's the source of the problem as we do receive bids from all our bidders, and the DFP setup should also be okay. The `adUnits` `code` is the first parameter of `googletag.defineSlot()` which also includes the name of the `div` that is also the parameter of `googletag.display()`.

Sooo I still don't have a clue of why this doesn't work... Thanks again for your feedback though!
